spec/integration/user_post_index_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+RSpec.describe 'User Posts', type: :system do
+  describe 'index page' do
+    before :each do
+      @user = User.create(name: 'micheal', photo: 'allamurotov@mail.ru', bio: 'bio')
+      @post = Post.create(user_id: @user.id, title: 'title', text: 'text')
+      @comment = Comment.create(user_id: @user.id, post_id: @post.id, text: 'some text')
+      visit user_posts_path(@user.id)
+    end
+
+    it "can see the user's profile picture" do
+      expect(page.find('.img_placeholder1')['alt']).to eq('some img')
+    end
+
+    it "can see user's username" do
+      user = User.first
+      expect(page).to have_text(user.name)
+    end
+
+    it 'can see the number of posts the user has written' do
+      user = User.first
+      expect(page).to have_text(user.posts.count)
+    end
+
+    it "can see a post's title" do
+      @user.posts.each do |i|
+        expect(i.title).to eq('title')
+      end
+    end
+
+    it "can see some of the post's body" do
+      user = User.first
+      user.posts.each do |i|
+        expect(page).to have_text(i.text)
+      end
+    end
+
+    it 'can see a first comment on a post' do
+      expect(page).to have_text(@comment.text)
+    end
+
+    it 'can see how many comments a post has' do
+      post = Post.first
+      expect(page).to have_text(post.comments.count)
+    end
+
+    it 'can see how many likes a post has' do
+      post = Post.first
+      expect(page).to have_text(post.likes.count)
+    end
+
+    it "When I click on a post, it redirects me to that post's show page" do
+      click_link 'Post 1'
+      expect(page).to have_current_path('/users/25/posts/25')
+    end
```

Comment:
- [ ] As the previous reviewer mentioned, Please add a test that asserts you can see a section for pagination if there are more posts than fit on the view.

![image](https://user-images.githubusercontent.com/68796232/184107368-8e21fb34-214a-4f86-81a4-0ac2a64351ae.png)

Reply:
I don't know how to implement that. could you throw some light?

Reply:
Hi @Gnwin,
You suppose to add the `pagination` button first to your `post#index` [view](https://github.com/Gnwin/blogapp/tree/f/integration-specs-n%2B1-queries/app/views/posts/index.html.erb) first because you didn't implement it yet, then kindly add the test for it.